pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,489 @@
+            "single": "T3",
+            "hold": "T4",
+            "fast": "T5",
+            }
+
+    ERRORS={
+            1: "Calibration error",
+            2: "RAM error",
+            4: "ROM error",
+            8: "AD slope error",
+            16: "AD converter error",
+            32: "AD link error",
+           }
+    
+
+
+    #decoder functions
+    @classmethod
+    def decode_status(self,status_bytes, field=None):
+        """Method to handle the decoding of the status bytes into something meaningfull
+
+        :param status_bytes:    list of bytes to be decoded
+        :return ret_val:
+
+        """
+        ret_val = Status(Status_bytes(*status_bytes))
+        if field is None:
+            return ret_val.b
+        elif field == "SRQ":
+            return getattr(ret_val.B, "byte3")
+        else:
+            return getattr(ret_val.b, field)
+
+    @classmethod
+    def decode_mode(cls,status_bytes):
+        """Method to decode current mode
+
+        :param status_bytes:   list of bytes to be decoded
+        :return cur_mode: string with the current trigger mode
+        :rtype cur_mode: str
+
+        """
+        cur_stat = Status(Status_bytes(*status_bytes))
+        function = cur_stat.b.function
+        inv_modes = {v: k for k, v in HP3478A.MODES.items()}
+        cur_mode = inv_modes["F"+str(function)]
+        return cur_mode
+
+    @classmethod
+    def decode_range(cls,status_bytes):
+        """Method to decode current range
+
+        :param status_bytes:   list of bytes to be decoded
+        :return cur_range: string with the current trigger mode
+        :rtype cur_range: float
+
+        """
+        cur_stat = Status(Status_bytes(*status_bytes))
+        function = cur_stat.b.function
+        inv_modes = {v: k for k, v in HP3478A.MODES.items()}
+        cur_mode = inv_modes["F"+str(function)]
+        rnge = cur_stat.b.range
+        cur_range = list(HP3478A.RANGES[cur_mode].keys())[rnge-1]
+        return cur_range
+
+    @classmethod
+    def decode_trigger(cls,status_bytes):
+        """Method to decode trigger mode
+
+        :param status_bytes:   list of bytes to be decoded
+        :return trigger_mode: string with the current trigger mode
+        :rtype trigger_mode: str
+
+        """
+        cur_stat= Status(Status_bytes(*status_bytes))
+        i_trig = cur_stat.b.int_trig
+        e_trig = cur_stat.b.ext_trig
+        if i_trig == 0:
+            if e_trig == 0:
+                trigger_mode = "hold"
+            else:
+                trigger_mode = "external"
+        else:
+            trigger_mode = "internal"
+        return trigger_mode
+
+    #commands/properties for instrument control
+    active_connectors = Instrument.measurement(
+        "B",
+        """Return selected connectors ("front"/"back"), based on front-panel selector switch
+        """,
+        get_process = (lambda x: HP3478A.decode_status(x.encode(encoding="ASCII"), "front_rear")),
```

Comment:
As you always use `x.encode(....)`, I think it would be nicest if you put the `encode` into the `decode_..` methods as well

Reply:
Will do.
Imrproves the readabibility of the code